### PR TITLE
Basic pin mapping for RedBear Blend2

### PR DESCRIPTION
**Pull request by pctj101**
Based mostly on BSP for rb-nano2.

Updated pin mapping for UART0
- LED Pin remapped
- SPI0 remapped/enabled
- I2C-0 remapped/enabled
- UART1/TWI-1/SPIM-1  - remapped/disabled due to pin conflicts

### CODE REVIEW

**Comment by wes3** [comment]
The pull request looks good except for one item: it appears you have mapped the LED_BLINK_PIN (P25) to both the LED and SPI0 SCK pin. I took a look at the schematic and P25 does seem to be the LED pin (although not sure what that MCP6002T chip is) so you will need to map SCK to some other pin. Note that I do not have a blend2 board with which to test and admittedly I did not go through every line of the .ld files and scripts so am taking some of this on faith :-)


To answer your questions:
1) Best way to make UART1 optional is to go into the syscfg.yml file and change UART_1 to value: 0
2) Looking at the schematic it appears that there are two sets of pins for i2c: SDA1/SCL1 (pins 26/27) and SDA/SCL (pins 13/12). So I would map i2c to those pins. None of those pins appear mapped already in your syscfg or hal_bsp.
3) I would not map SPI1 personally. There are not a ton of GPIO that go out to the connectors. If folks need another SPI they should just map it in their BSP.

**Comment by pctj101** [comment]
Believe it or not... P25 is "officially" shared with SPI0_SCK
https://github.com/redbear/nRF5x#blend-2

Should we follow the "official" image, or... do you think we should make a command decision to make something more... "useful"?

Looking into the other items...

**Comment by pctj101** [comment]
Wanted to post an update in regards to your feedback.  

I added I2C_0 to syscfg/hal_bsp

I left UART_1 configured as value:1 because "most arduino shaped boards like the Blend2" have serial on those pins.

I'll post another note when I have time to test the new pin mappings.

**Comment by pctj101** [comment]
Regarding the .ld/script files - since the rb-nano2 and rb-blend2 share the same "Redbear MB-N2" module, it's quite likely that the rb-nano2 settings can simply be used for the rb-blend2.

**Comment by wes3** [comment]
I would not share the LED with SCK personally. There are other pins to use and you can use any pin for any peripheral so keeping them separate makes sense to me

File: hw/bsp/rb-blend2/include/bsp/boards.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/*
```
Comment (utzig) [inline hw/bsp/rb-blend2/include/bsp/boards.h]: Is this file supposed to be only a license header?

Comment (pctj101) [inline hw/bsp/rb-blend2/include/bsp/boards.h]: Ah I blindly copied it from rb-nano2... so I meant nothing other than the license comment already included in this file.

**Comment by pctj101** [comment]
@wes3 You know what's crazy? In the Arduino Uno pinout, SCK and LED are actually officially shared.

Does MyNewt have any intention of following the Arduino Uno pinout for boards like the Blend2 which are based on the Arduino Uno pinout?

**Comment by wes3** [comment]
I cannot speak for all of Mynewt obviously but I think folks generally try to allocate the pins such that there is one i2c interface, one spi interface, and the LEDs are not shared. I see no reason to commit a bsp that shares any of these unless there is not enough gpio. I do not think that is the case with any of the boards you mention (although I could be wrong).

**Comment by pctj101** [comment]
@wes3 Here's some more info for both sides of the coin:

http://playground.arduino.cc/Main/ShieldPinUsage
Any correctly designed shield will connect the SPI lines(MISO, MOSI, SCK) via the ICSP header instead of the standard Arduino I/O pin headers because the SPI is found on different pins depending on the board.

But.... the Blend2 board (in the pinning for the Arduino Uno) doesn't have an ICSP header, so.......

Would it be better to at least have the Arduino Uno standard available? (not a super strong argument in light of even Arduino saying "don't do it")

Anyways, this is the only reason I can find to keep SCK and LED on the same pin.

Otherwise, I totally agree with you.

**Comment by pctj101** [comment]
I'm totally out of good reasons to support keeping with the Arduino Uno SPI standard especially when Arduino says don't do it.  

So I picked the next nearest pin which is unassigned.

Is there a suggested place to document this "deviation" from the official "RedBear Blend2 pinout diagram" so that new users aren't surprised?

**Comment by wes3** [comment]
You could submit a tutorial here that describes getting the blend2 up and running: http://mynewt.incubator.apache.org/latest/os/introduction/

Otherwise a comment in the code is fine :-)

**Comment by wes3** [comment]
It looks good; commit has been merged.

**Comment by utzig** [comment]
@pctj101 Please close this issue, the PR was already merged but the issue was not automatically closed. Thanks!

**Pull request closed without merging**